### PR TITLE
Anonymous cross origin for images

### PR DESCRIPTION
**Pull request by Javasamurai**

### CODE REVIEW

File: src/ui/resource/Image.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ exports = Class(lib.PubSub, function () {
     // create an image if we don't have one
     if (!img) {
       img = new Image();
-      img.crossOrigin = 'use-credentials';
```
Comment (rampr) [inline src/ui/resource/Image.js:150]: I don't we should change it like this. Mobile might break.

**Review by rampr — CHANGES_REQUESTED** [review]

File: src/ui/resource/Image.js (hunk 1 above)
Comment (Javasamurai) [reply src/ui/resource/Image.js:150]: Checked on ios with no issues.

Comment (rampr) [reply src/ui/resource/Image.js:150]: what is the purpose of use-credentials ?

**Comment by rampr** [comment]
As discussed, let's pass `crossOrigin` in `opts` and if `opts.crossOrigin` exists we set that.
Something like

`img.crossOrigin = opts.crossOrigin? opts.crossOrigin: 'use-credentials';`

File: src/ui/resource/Image.js [hunk 2]
```diff
@@ -147,7 +148,9 @@ exports = Class(lib.PubSub, function () {
...
+      if (!this._cors_enabled) {
+        img.crossOrigin = 'anonymous';
```
Comment (rampr) [inline src/ui/resource/Image.js]: if you set the value from opts.cors, the caller will have the freedom to set it to whatever they want.
Right now use-credentials is never being set.

File: src/ui/resource/Image.js (hunk 1 above)
Comment (Javasamurai) [reply src/ui/resource/Image.js:150]: For user-authentication using CORS.

File: src/ui/resource/Image.js (hunk 2 above)
Comment (Javasamurai) [reply src/ui/resource/Image.js]: Updated for that.

**Review by rampr — APPROVED** [review]

**Pull request merged**